lib/PuppeteerSharp.Tests/Page/GotoTests.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace PuppeteerSharp.Tests.Page
+{
+    [Collection("PuppeteerLoaderFixture collection")]
+    public class GotoTests : PuppeteerPageBaseTest
+    {
+        [Fact]
+        public async Task ShouldNavigateToAboutBlank()
+        {
+            var response = await Page.GoToAsync(TestConstants.AboutBlank);
+            Assert.Null(response);
+        }
+
+        [Theory]
+        [InlineData("domcontentloaded")]
+        [InlineData("networkidle0")]
+        [InlineData("networkidle2")]
+        public async Task ShouldNavigateToEmptyPage(string waitUntil)
+        {
+            var response = await Page.GoToAsync(TestConstants.EmptyPage, new NavigationOptions { WaitUntil = new[] { waitUntil } });
+            Assert.Equal(HttpStatusCode.OK, response.Status);
+        }
+
+        [Fact]
+        public async Task ShouldFailWhenNavigatingToBadUrl()
+        {
+            var exception = await Assert.ThrowsAnyAsync<Exception>(async () => await Page.GoToAsync("asdfasdf"));
+            Assert.Contains("Cannot navigate to invalid URL", exception.Message);
+        }
+
+        [Fact(Skip = "message is ERR_CERT_COMMON_NAME_INVALID instead of ERR_CERT_AUTHORITY_INVALID")]
```

Comment:
I think these errors are caused by the server not returning the correct error

Reply:
Just leaving this here so we have some reference https://stackoverflow.com/questions/44711260/chorme-59-neterr-cert-common-name-invalid